core/modules/ccontrol/MergingHandler.cc

```diff
@@ -248,11 +242,11 @@ void MergingHandler::_setError(int code, std::string const& msg) {
     _error = Error(code, msg);
 }
 
-bool MergingHandler::_setResult(BufPtr const& bufPtr) {
+bool MergingHandler::_setResult(BufPtr const& bufPtr, int blen) {
     auto start = std::chrono::system_clock::now();
     std::lock_guard<std::mutex> lg(_setResultMtx);
     auto& buf = *bufPtr;
-    if (!ProtoImporter<proto::Result>::setMsgFrom(_response->result, &(buf[0]), buf.size())) {
+    if (!ProtoImporter<proto::Result>::setMsgFrom(_response->result, &(buf[0]), blen)) {
```

Comment:
Since I don't see what BufPtr is now it clearly is no longer a container, right? So, I assume it needs an external blen. Any reason for that change?

Reply:
The result for this message and the header for the next message are in one buffer. The header is small, so it's just copied to another buffer. The result could be big, so it stays in the buffer and blen is the length of the buffer minus the header.